lms/services/assignment.py

```diff
@@ -11,225 +9,48 @@ class AssignmentService:
     def __init__(self, db):
         self._db = db
 
-    @lru_cache(maxsize=128)
-    def get(
-        self,
-        tool_consumer_instance_guid,
-        resource_link_id=None,
-        ext_lti_assignment_id=None,
-    ):
-        """Get an assignment using using resource_link_id, ext_lti_assignment_id or both."""
-
-        if not any([resource_link_id, ext_lti_assignment_id]):
-            raise ValueError(
-                "Can't get an assignment without neither resource_link_id or ext_lti_assignment_id"
-            )
-
-        if all([resource_link_id, ext_lti_assignment_id]):
-            # When launching an assignment in Canvas there are both resource_link_id and
-            # ext_lti_assignment_id launch params.
-            assignments = self.get_for_canvas_launch(
-                tool_consumer_instance_guid, resource_link_id, ext_lti_assignment_id
-            )
-            if len(assignments) > 1:
-                raise MultipleResultsFound(
-                    "Multiple assignments found. Should merge_canvas_assignments have been called"
-                )
-            return assignments[0] if assignments else None
-
-        if ext_lti_assignment_id:
-            # When creating or editing an assignment Canvas launches us with an
-            # ext_lti_assignment_id but no resource_link_id.
-            return self._get_for_canvas_assignment_config(
-                tool_consumer_instance_guid, ext_lti_assignment_id
-            )
-
-        # Non-Canvas assignment launches always have a resource_link_id and never
-        # have an ext_lti_assignment_id.
-        return self._get_by_resource_link_id(
-            tool_consumer_instance_guid, resource_link_id
-        )
-
-    @lru_cache(maxsize=128)
-    def get_for_canvas_launch(
-        self,
-        tool_consumer_instance_guid,
-        resource_link_id,
-        ext_lti_assignment_id,
-    ):
-        """
-        Return the assignment(s) with resource_link_id or ext_lti_assignment_id.
-
-        Return all the assignments in the DB that have the given tool_consumer_instance_guid and
-        either the given resource_link_id or ext_lti_assignment_id or both. This could be:
-
-        1. A single assignment in the DB that has either the given resource_link_id or
-           ext_lti_assignment_id or both
-
-        2. Or two assignments:
-
-           i.  One with the matching resource_link_id and no ext_lti_assignment_id
-           ii. And one with the matching ext_lti_assignment_id and no resource_link_id
-
-           The assignment with the resource_link_id will always be first in the sequence.
-
-        :rtype: sequence of either 0, 1 or 2 models.Assignment objects
-        """
-        assert resource_link_id and ext_lti_assignment_id
+    def get(self, tool_consumer_instance_guid, resource_link_id):
```

Comment:
All of the code here (in this file) deals mostly with:

- Finding things by either of two different IDS.
- Merging assignments when they were created at different times with different IDs.